scripts/run-affected-tests.sh

```diff
@@ -0,0 +1,50 @@
+#!/bin/bash
+set -e
+
+# Script to run tests only for packages affected by changes
+# Used in CI to optimize test execution time
+
+BASE_BRANCH="${1:-main}"
+echo "Analyzing packages affected by changes from $BASE_BRANCH..."
+
+# Use turbo dry-run to identify which packages need testing
+# This includes both changed packages and their dependents
+TURBO_OUTPUT=$(yarn turbo run test:run --filter="./packages/*" --filter="[origin/$BASE_BRANCH...HEAD]" --dry-run 2>&1)
+
+# Extract directories of packages that have test:run tasks
+# 1. Find lines like "@alchemy/package-name#test:run"
+# 2. Look ahead to find the Directory line
+# 3. Extract the directory path and remove the "packages/" prefix
+AFFECTED_DIRS=$(echo "$TURBO_OUTPUT" | \
```

Comment:
this feels like it might be a little brittle (i.e. if turbo changes the log output). fine w/ it for now, but not sure if it's worth revisiting later.

Reply:
yeah agreed. will keep an eye out. worst case, we just revert to what we had before with the standard test run suite.